File: Cadena de Hoteles/Hotel/src/main/java/org/example/helpers/TokenHelper.java
package org.example.helpers;

import java.security.SecureRandom;
import java.util.HexFormat;

/**
 * Helper para la generacion de tokens de autenticacion seguros.
 * Usa SecureRandom para garantizar aleatoriedad criptografica.
 */
public class TokenHelper {

    /**
     * Genera un token aleatorio de 64 caracteres en formato hexadecimal.
     * Se usa como token de acceso para agencias externas u otros casos
     * donde se necesite un identificador unico e impredecible.
     *
     * @return string hexadecimal de 64 caracteres generado con SecureRandom.
     */
    public static String generarTokenHash() {
        byte[] bytes = new byte[32];
        new SecureRandom().nextBytes(bytes);
        return HexFormat.of().formatHex(bytes); // 64 chars hex
    }
}

File: Aero Linea/Aerolinea.API/Aerolinea.API/Helpers/TokenHelper.cs
using System.Security.Cryptography;

namespace Aerolinea.API.Helpers
{
    /// <summary>
    /// Clase estatica de utilidad para la generacion de tokens seguros y unicos.
    /// Utiliza el generador de numeros aleatorios criptograficos del sistema
    /// para producir tokens con alta entropia aptos para autenticacion de agencias,
    /// restablecimiento de contrasena u otros flujos que requieran tokens opacos.
    /// </summary>
    public static class TokenHelper
    {
        /// <summary>
        /// Genera un token unico y seguro de 64 caracteres hexadecimales en minusculas.
        /// Internamente obtiene 32 bytes aleatorios criptograficamente seguros,
        /// les aplica SHA-256 y convierte el resultado a cadena hexadecimal.
        /// </summary>
        public static string GenerarTokenHash()
        {
            var bytes = RandomNumberGenerator.GetBytes(32);
            var hash = SHA256.HashData(bytes);
            return Convert.ToHexString(hash).ToLower();
        }
    }
}
